src/shared/mount-util.c

```diff
@@ -1049,14 +1049,42 @@ int make_mount_point(const char *path) {
         return 1;
 }
 
-static int make_userns(uid_t uid_shift, uid_t uid_range) {
-        char line[DECIMAL_STR_MAX(uid_t)*3+3+1];
+/* If REMOUNT_IDMAP_HOST_ROOT is set we'll include a mapping here that maps the host root user accessing the
+ * idmapped mount to the this user ID on the backing fs. This is the last valid UID in the *signed* 32bit
+ * range. You might wonder why precisely use this specific UID for this purpose? Well, we definitely cannot
+ * use the first 0…65536 UIDs for that, since in most cases that's precisely the file range we intend to map
+ * to some high UID range, ans since UID mappings have to be bijective we thus cannot use them at
+ * all. Furthermore the UID range beyond INT32_MAX (i.e. the range above the signed 32bit range) is icky,
+ * since many APIs cannot use it (example: setfsuid() returns the old UID as signed integer). Following our
+ * usual logic of assigning a 16bit UID range to each container, so that the upper 16bit of a 32bit UID value
+ * indicate kind of a "container ID" and the lower 16bit map directly to the intended user you can read this
```

Comment:
should this be documented in one of our interfaces in docs/ ?

Reply:
For now I'd consider this an internalism of nspawn. files created this way will only be owned by this UID for a very brief time, because we first create the inodes and then chown them. Other tools should never see them